index.js

```diff
@@ -278,7 +278,7 @@ function runCmd_ (cmd, pkg, env, wd, opts, stage, unsafe, uid, gid, cb_) {
   proc.on('close', (code, signal) => {
     opts.log.silly('lifecycle', logid(pkg, stage), 'Returned: code:', code, ' signal:', signal)
     let err
-    if (signal) {
+    if (signal && signal !== "SIGINT") {
       err = new PnpmError('CHILD_PROCESS_FAILED', `Command failed with signal "${signal}"`)
       process.kill(process.pid, signal)
```

Comment:
ok, it shouldn't throw an error but shouldn't it still kill the main process with the same signal?

Shouldn't `process.kill(process.pid, signal)` still be executed?

Reply:
The child process may have interrupted, but why would `pnpm` be? The parent process simply terminates since the child ended by a user (normally) signal.